src/particle.cpp

```diff
@@ -353,6 +358,40 @@ void Particle::event_death()
   }
 }
 
+void Particle::energy_delivert_in_cell()
+{
+  // Update pht_storage array
+  pht_storage()[coord(0).cell] += E_last() - E();
+}
+
+void Particle::killed_particle_energy_delivert()
+{
+  // Add the energy of the killed particles to the pht_storage array
+  pht_storage()[coord(0).cell] += E();
+}
+
+void Particle::remove_energy_of_secondary()
+{
+
+  // determine where the particle was born
+  if (coord(n_coord() - 1).cell == C_NONE) {
+    if (!exhaustive_find_cell(*this)) {
+      mark_as_lost(
+        "Could not find the cell containing particle " + std::to_string(id()));
+      return;
+    }
+
+    // Set birth cell attribute
+    if (cell_born() == C_NONE)
```

Comment:
Why do you have to do this here? I think you are probably calling this function a bit early somewhere if you are making yourself responsible for setting the `cell_born()` attribute. I am guessing that adding this fixed a bug for you at some point. I think we could probably remove this, but I'll have to look into it more to be sure.

Reply:
I replaced 
```cpp
coord(0).cell
```
to make the tally work on multiple leveled universes
```cpp
coord(n_coord() - 1).cell
```